File: src/main/java/com/bdev/hogwarts_api/data/model/student/StudentPhoneModel.java
package com.bdev.hogwarts_api.data.model.student;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "HG_STUDENT_PHONE")
public class StudentPhoneModel {
    @Id
    @Column(name = "ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STUDENT_ID", nullable = false)
    private StudentModel student;

    @Column(name = "VALUE", length = 256)
    private String value;
}

File: src/main/java/com/bdev/hogwarts_api/data/model/student/StudentPhoneModel.kt
package com.bdev.hogwarts_api.data.model.student

import lombok.Data

import javax.persistence.*

@Data
@Entity
@Table(name = "HG_STUDENT_PHONE")
open class StudentPhoneModel {
    @Id
    @Column(name = "ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STUDENT_ID", nullable = false)
    var student: StudentModel? = null

    @Column(name = "VALUE", length = 256)
    var value: String = ""
}
